packages/js/src/notifications/notifications.ts

```diff
@@ -21,28 +31,48 @@ import type {
   BaseArgs,
 } from './types';
 import { NovuError } from '../utils/errors';
+import { NotificationsCache } from '../cache';
 
 export class Notifications extends BaseModule {
-  async list({ limit = 10, ...restOptions }: ListNotificationsArgs = {}): Result<ListNotificationsResponse> {
+  readonly #notificationsCache: NotificationsCache;
+
+  constructor() {
+    super();
+    this.#notificationsCache = new NotificationsCache();
+  }
+
+  async list(
+    { limit = 10, ...restOptions }: ListNotificationsArgs = {},
+    options: { cache: boolean } = { cache: true }
+  ): Result<ListNotificationsResponse> {
     return this.callWithSession(async () => {
       const args = { limit, ...restOptions };
       try {
-        this._emitter.emit('notifications.list.pending', { args });
-
-        const response = await this._inboxService.fetchNotifications({
-          limit,
-          ...restOptions,
-        });
-
-        const modifiedResponse: ListNotificationsResponse = {
-          hasMore: response.hasMore,
-          filter: response.filter,
-          notifications: response.data.map((el) => new Notification(el)),
-        };
+        let data: ListNotificationsResponse | undefined = options.cache
+          ? this.#notificationsCache.getAll(args)
+          : undefined;
+
+        if (!data) {
+          const response = await this._inboxService.fetchNotifications({
+            limit,
+            ...restOptions,
+          });
+
+          data = {
+            hasMore: response.hasMore,
+            filter: response.filter,
+            notifications: response.data.map((el) => new Notification(el)),
+          };
+
+          if (options.cache) {
+            this.#notificationsCache.set(args, data);
+            data = this.#notificationsCache.getAll(args);
+          }
```

Comment:
After we get a response from the server, store the data in the cache and return aggregated notifications.